sepal_ui/sepalwidgets/widget.py

```diff
@@ -126,3 +127,62 @@ def _clip(self, widget, event, data):
         self.tf.append_icon = "fas fa-clipboard-check"
 
         return
+
+
+class StateIcon(Tooltip):
+    """
+    Custom icon with multiple state colors.
+
+    Args:
+        model (sepal_ui.Model): Model to manage StateIcon behaviour from outside.
+        model_trait (str): Name of trait to be linked with state icon. Must exists in model.
+        states (dict): Dictionary where keys are the state name to be linked with self value and value represented by a tuple of two elements. {"key":(tooltip_msg, color)}.
+    """
+
+    value = Unicode().tag(sync=True)
+    "str: key name of the current state of component. Values must be same as states_dict keys."
+
+    states = None
+    'dict: Dictionary where keys are the state name to be linked with self value and value represented by a tuple of two elements. {"key":(tooltip_msg, color)}.'
+
+    right = True
+    "bool: TODO"
+
+    icon = None
+    "v.Icon: the colored Icon of the tooltip"
+
+    def __init__(self, model, model_trait, states=None):
+
+        default_states = {
+            "valid": ("Valid", color.success),
+            "non_valid": ("Not valid", color.error),
+        }
+
+        self.states = default_states if not states else states
+
+        self.right = True
```

Comment:
what is the purpose of this attribute ? It's set to True at init level and then never changed

Reply:
https://github.com/12rambau/sepal_ui/pull/469/commits/4756607c08c4513b1e128e8c880d7231cb63c4cb